modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements/evidence_submissions_controller.rb

```diff
@@ -15,6 +16,16 @@ def show
 
           render json: serialized.serializable_hash
         end
+
+        def create
+          status, error = AppealsApi::FileValidator.new(params[:document]).call
+
+          if status == :ok
+            render json: { document: params[:document], uuid: params[:uuid] }
+          else
+            raise Common::Exceptions::UnprocessableEntity.new(detail: error)
```

Comment:
The error we're getting back from the validator is roughly equivalent to a JSON API ErrorObject already. The `detail` of an ErrorObject (which the Common::Exceptions generally are, IIRC) is supposed to be a human-readable string, so putting the `error` object into it breaks ErrorObject norms. The `error` should already have the `title`, `detail`, and `meta` that we can then mix into an error array. See how DetailedSchemaErrors builds up a hash to turn into a final response - this controller should do the same.

Reply:
I feel like we can do something much more simple, actually. Check out my solution, yo!